### PR TITLE
FreeBSD: enable spa_thread

### PR DESCRIPTION
**Pull request by mattmacy**
Avoid a cosmetic regression when downstreaming.
Bind each pool to its own thread, put the pool
name in the proc title, etc.

Signed-off-by: Matt Macy <mmacy@FreeBSD.org>

### How Has This Been Tested?

Checked `ps ax` for zpool-$poolname, ran some cli_root tests to shake

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Performance enhancement (non-breaking change which improves efficiency)
- [ ] Code cleanup (non-breaking change which makes code smaller or more readable)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation (a change to man pages or other documentation)

### Checklist:

- [ ] My code follows the ZFS on Linux [code style requirements](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#coding-conventions).
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**contributing** document](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md).
- [ ] I have added [tests](https://github.com/zfsonlinux/zfs/tree/master/tests) to cover my changes.
- [ ] I have run the ZFS Test Suite with this change applied.
- [ ] All commit messages are properly formatted and contain [`Signed-off-by`](https://github.com/zfsonlinux/zfs/blob/master/.github/CONTRIBUTING.md#signed-off-by).

### CODE REVIEW

File: module/zfs/spa.c [hunk 1]
```diff
@@ -1097,54 +1097,20 @@ spa_create_zio_taskqs(spa_t *spa)
 /*
  * Disabled until spa_thread() can be adapted for Linux.
  */
-#undef HAVE_SPA_THREAD
```
Comment (behlendorf) [inline module/zfs/spa.c:1100]: Can you update this comment accordingly.

**Comment by freqlabs** [comment]
We're taking a look at some test failures in send/recv with this patch.

**Comment by kev009** [comment]
@freqlabs on a lark, could you re-test this with https://github.com/openzfs/zfs/pull/12350 applied?

**Comment by freqlabs** [comment]
Sure, I've been wanting to revisit this.

**Comment by freqlabs** [comment]
Hrmm I thought reopening and pushing would run the CI again but I'm not seeing it.

**Comment by freqlabs** [comment]
Still failing:
```
Tests with results other than PASS that are unexpected:
    FAIL checksum/filetest_001_pos (expected PASS)
    FAIL checksum/filetest_002_pos (expected PASS)
    FAIL cli_root/zfs_receive/zfs_receive_from_zstd (expected PASS)
    FAIL cli_root/zfs_receive/zfs_receive_raw (expected PASS)
    FAIL cli_root/zfs_receive/zfs_receive_raw_incremental (expected PASS)
    FAIL cli_root/zfs_rename/zfs_rename_002_pos (expected PASS)
    FAIL cli_root/zfs_rename/zfs_rename_004_neg (expected PASS)
    FAIL cli_root/zfs_rename/zfs_rename_005_neg (expected PASS)
    FAIL removal/removal_remap_deadlists (expected PASS)
    FAIL rsend/send_encrypted_props (expected PASS)
    FAIL rsend/send_freeobjects (expected PASS)
    FAIL rsend/send_mixed_raw (expected PASS)
```

**Comment by freqlabs** [comment]
Oh sorry I need to test again with that commit cherry-picked.

**Comment by freqlabs** [comment]
With HAVE_LARGE_STACKS things did go much better, at least for FreeBSD.

**Pull request closed without merging**